### PR TITLE
Repository::update_at with tests

### PR DESCRIPTION
**Pull request by zie1ony**

### CODE REVIEW

File: tests/lib.rs [hunk 1]
```diff
@@ -1,2 +1,4 @@
+#![feature(once_cell)]
```
Comment (kpob) [inline tests/lib.rs:1]: What does it do?

File: utils/src/modules/repository.rs [hunk 2]
```diff
@@ -32,19 +36,43 @@ impl Repository {
     }
 
     pub fn update_at(&mut self, key: String, value: Bytes, activation_time: Option<u64>) {
-        let mode = Repository::resolve_update_mode(activation_time);
-        let (current, _) = self.storage.get(&key);
-        let new_value = match mode {
-            UpdateMode::SetFuture => (current, Some((value.clone(), activation_time.unwrap()))),
-            UpdateMode::ClearFuture => (current, None),
-            UpdateMode::Current => (value.clone(), None),
+        let now = get_block_time();
```
Comment (kpob) [inline utils/src/modules/repository.rs:39]: `get()` method should be refactored as well to approach the block time the same way

File: utils/src/modules/repository.rs [hunk 3]
```diff
@@ -32,19 +36,43 @@ impl Repository {
...
+        let value_for_event = value.clone();
+        let new_value: Record = match activation_time {
```
Comment (kpob) [inline utils/src/modules/repository.rs:41]: It's really hard to read this `match` statement. I feel lost especially in the last complex `Some(activation_time)` arm. Btw. I wasn't aware of the Some if syntax. I liked the approach of resolving cases in a separate function. Then the main `update_at()` simply sets values. I know, comments describe each case, but I prefer simple functions, all the complex logic can be hidden.
What do think?

**Review by kpob — APPROVED** [review]

File: contracts/src/variable.rs [hunk 4]
```diff
@@ -112,6 +124,7 @@ impl VariableRepositoryContractTest {
         let count: u32 = self
             .env
             .get_value(self.package_hash, self.data.repository.keys.length.path());
+        dbg!(count);
```
[the remark this answers is not in the document]
Comment (zie1ony) [reply contracts/src/variable.rs:127]: no

File: tests/lib.rs (hunk 1 above)
Comment (zie1ony) [reply tests/lib.rs:1]: I was experimenting with Lazy. I'll remove it, as that didn't get me anywhere.

File: utils/src/modules/repository.rs (hunk 3 above)
Comment (zie1ony) [reply utils/src/modules/repository.rs:41]: Let's merge this one and you can propose your version.

File: utils/src/modules/repository.rs (hunk 2 above)
Comment (zie1ony) [reply utils/src/modules/repository.rs:39]: done

**Pull request merged**